iosapp/app/App.swift

```diff
@@ -1,182 +1,436 @@
-              await send(.setFirstLaunch(firstLaunch))
-            } else {
-              let now = self.now
-              self.storage.saveFirstLaunchDate(now)
-              await send(.setFirstLaunch(now))
-              await self.api.logEvent(
-                "dcd721aa",
-                "first launch, region: `\(self.locale.region?.identifier ?? "(nil)")`"
-              )
-            }
-          }
-        )
+  var body: some View {
+    switch self.store.screen {
+    case .launching:
+      EmptyView()
 
-      case .setRunning(true):
-        state.appState = .running(showVendorId: false)
-        return .none
-
-      case .setFirstLaunch(let date):
-        state.firstLaunch = date
-        return .none
-
-      case .setRunning(false):
-        state.appState = .welcome
-        return .none
-
-      case .welcomeNextTapped:
-        state.appState = .prereqs
-        return .none
-
-      case .startAuthorizationTapped:
-        state.appState = .authorizing
-        return .run { send in
-          switch await self.systemExtension.requestAuthorization() {
-          case .success:
-            await send(.authorizationSucceeded)
-            await self.api.logEvent("d317c73c", "authorization succeeded")
-          case .failure(let reason):
-            await send(.authorizationFailed(reason))
-            await self.systemExtension.cleanupForRetry()
-            await self.api.logEvent("d9dfd021", "authorization failed: \(reason)")
-          }
-        }
+    case .onboarding(.happyPath(.hiThere)):
+      WelcomeView {
+        self.store.send(.interactive(.onboardingBtnTapped(.primary, "Next")))
+      }
+      .onShake {
+        #if DEBUG
+          self.store.send(.interactive(.receivedShake))
+        #endif
+      }
 
-      case .authorizationSucceeded:
-        state.appState = .authorized
-        return .none
-
-      case .authorizationFailed(let reason):
-        state.appState = .authorizationFailed(reason)
-        return .none
-
-      case .authorizationFailedReviewRequirementsTapped:
-        state.appState = .prereqs
-        return .none
-
-      case .authorizationFailedTryAgainTapped:
-        state.appState = .welcome
-        return .none
-
-      case .installFilterTapped:
-        return .run { send in
-          switch await self.systemExtension.installFilter() {
-          case .success:
-            await send(.installSucceeded)
-            await self.api.logEvent("101c91ea", "filter install success")
-          case .failure(let error):
-            await send(.installFailed(error))
-            await self.systemExtension.cleanupForRetry()
-            await self.api.logEvent("739c08c6", "filter install failed: \(error)")
-          }
-        }
+    case .onboarding(.happyPath(.timeExpectation)):
+      ButtonScreenView(
+        text: "The setup usually takes about 5-7 minutes, but in some cases extra steps are required.",
+        primary: self.btn(text: "Next", .primary)
```

Comment:
this is a little abstraction i made to reduce some boilerplate around these buttons